### PR TITLE
Dynamic command management for notebook-only commands

### PR DESCRIPTION
**Pull request by jjallaire**
Some commands will be notebook-only (e.g. Restart R and Clear Output) and these commands will therefore need be dynamically managed based on whether inline output is enabled for the active document.

This commit introduces a scheme for this, in summary:

1. The active editor now gets a chance to manage commands after all default command management has taken place.

2. The notebook helper uses this hook to manage notebook-only commands based on the per-document and global preference.

3. The notebook helper now listens for changes in the global inline output preference and syncs it's own behavior accordingly (this change is orthagonal to the PR but was an ommission which needed correcting)

4. If the user interactively changes the inline output preference then commands are managed at that point as well.

@jmcphers

### CODE REVIEW

File: src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/rmd/TextEditingTargetNotebook.java [hunk 1]
```diff
@@ -228,6 +220,18 @@ public void initialize(EventBus events,
       events_.addHandler(ConsolePromptEvent.TYPE, this);
       events_.addHandler(InterruptStatusEvent.TYPE, this);
       events_.addHandler(RestartStatusEvent.TYPE, this);
+      
+      // subscribe to global rmd output inline preference and sync
+      // again when it changes
+      releaseOnDismiss.add(
```
Comment (jmcphers) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/rmd/TextEditingTargetNotebook.java]: Where does `releaseOnDismiss` unregister handlers?

Comment (jjallaire) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/rmd/TextEditingTargetNotebook.java]: Good catch! Fixed here: https://github.com/rstudio/rstudio/pull/649/commits/b843fe368ce1817c7c221fe3f5527a3cbd566e0b

**Comment by jmcphers** [comment]
LGTM. @jjallaire is this ready for merge?

**Comment by jjallaire** [comment]
It is indeed, I'll go ahead and merge it.

**Pull request merged**